.github/workflows/run_all_steps.yml

```diff
@@ -187,45 +187,80 @@ jobs:
-        run: tornettools generate
-          relayinfo_staging_*.json
-          userinfo_staging_*.json
-          networkinfo_staging.gml
-          tmodel-ccs2018.github.io
-          --network_scale 0.0001
-          --prefix tornet
+        run: |
+          # Minimum number of clients is 100; --torperf_scale=0.1 here should
+          # give us ~10 perf clients.
+          tornettools \
+            --seed 1 \
+            generate \
+            relayinfo_staging_*.json \
+            userinfo_staging_*.json \
+            networkinfo_staging.gml \
+            tmodel-ccs2018.github.io \
+            --network_scale 0.0001 \
+            --torperf_scale 0.1 \
+            --prefix tornet
 
       - name: Simulate
         run: |
           # Use short simulation time
-          sed -i 's/stop_time:.*/stop_time: "10m"/' tornet/shadow.config.yaml
+          sed -i 's/stop_time:.*/stop_time: "15m"/' tornet/shadow.config.yaml
           # Install runtime shadow and tor deps, in case we skipped building (and installing build deps)
           sudo apt-get install -y libevent-2.1-7 libssl1.1 zlib1g libglib2.0-0 libigraph0v5 libprocps8 sysstat python3 python3-pip
-          tornettools simulate --args '--parallelism=2 --seed=1 --template-directory=shadow.data.template --progress=true' tornet
+          tornettools simulate --args "--parallelism=$(nproc) --seed=1 --template-directory=shadow.data.template --progress=true" tornet
 
       - name: Parse
         # We use a relatively low convergence time (-c) since we're only
         # simulating 10m.
         run: tornettools parse -c 300 tornet
 
       - name: Plot
-        run: tornettools plot tornet --tor_metrics_path tor_metrics_2020-11-*.json --prefix pdfs
+        # Plot against saved artifacts from a previous run, for comparison.
+        run: tornettools plot --tor_metrics_path tor_metrics_*.json --prefix pdfs -l golden current -- .github/workflows/golden_tornet tornet
+
+      - name: Generate README
+        run: |
+          echo "Generated by run_all_steps workflow with:" > tornet/README
+          for v in \
+            GITHUB_REF \
+            GITHUB_SHA \
+            SHADOW_COMMIT \
+            TOR_REPO \
+            TOR_BRANCH \
+            TOR_COMMIT \
+            TGEN_COMMIT \
+            ONIONTRACE_COMMIT \
+            NETDATA_MONTH \
+            NETDATA_LAST_DAY \
+            ; do echo "$v: ${!v:-}" >> tornet/README; done
 
       - name: Archive
         run: tornettools archive tornet
 
+      # This artifact is intended to include exactly what ought to be checked
+      # in as a "golden" output. Files that don't need to be included there
+      # should go in separate artifacts.
+      - name: Upload tornet
+        uses: actions/upload-artifact@v2
+        with:
+          name: tornet
+          path: |
+            tornet/README
+            tornet/tornettools.*.log*
+            tornet/*.json.xz
+            tornet/tornet.plot.data/*.json
```

Comment:
This still seems a bit confusing.

`run_all_steps.yml` says:

> This artifact includes exactly what ought to be checked in as a "golden" output in `.github/workflows/golden_tornet`. To update the golden output, unpack this artifact into that directory.

`parse_and_plot.yml` says:

> When this test fails due to an *intentional* change in processing, the golden data should be updated using the actual output from this workflow's artifacts.

Each workflow says that the golden output/data should be replaced with data from different workflow artifacts.

Maybe it would be better to say explicitly for each step that could fail: "If this step fails, copy the "<artifact name>" artifact from workflow "<workflow name>" to the ".github/workflow/<...>" directory."

Reply:
Hopefully clearer now. I split `golden_tornet` into `run_all_steps_output`, `parse_and_plot_input`, and `parse_and_plot_output`.